server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,137 +265,441 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
-                    updateAllNewOnTier(tier, allLookups);
-                  } else {
-                    updateAllOnTier(tier, allLookups);
-                    deleteAllOnTier(tier, drops);
-                  }
-                }
-                catch (InterruptedException e) {
-                  Thread.currentThread().interrupt();
-                  throw Throwables.propagate(e);
-                }
-                catch (Exception e) {
-                  LOG.error(e, "Error updating lookups for tier [%s]. Will try again soon", tier);
-                }
-              }
-              prior_update = allLookupTiers;
             }
-          },
-          0,
-          lookupCoordinatorManagerConfig.getPeriod(),
-          TimeUnit.MILLISECONDS
-      );
-      Futures.addCallback(backgroundManagerFuture, new FutureCallback<Object>()
-      {
-        @Override
-        public void onSuccess(@Nullable Object result)
-        {
-          backgroundManagerExitedLatch.countDown();
-          LOG.debug("Exited background lookup manager");
+        );
+
+        configManager.set(
+            LOOKUP_CONFIG_KEY,
+            converted,
+            new AuditInfo("autoConversion", "autoConversion", "127.0.0.1")
+        );
+      }
+    }
+  }
+
+  private Map<String, LookupExtractorFactoryMapContainer> convertTierLookups(
+      Map<String, Map<String, Object>> oldTierLookups
+  )
+  {
+    Map<String, LookupExtractorFactoryMapContainer> convertedTierLookups = new HashMap<>();
+    oldTierLookups.forEach(
+        (lookup, lookupExtractorFactory) -> {
+          convertedTierLookups.put(lookup, new LookupExtractorFactoryMapContainer(null, lookupExtractorFactory));
         }
+    );
+    return convertedTierLookups;
+  }
 
-        @Override
-        public void onFailure(Throwable t)
-        {
-          backgroundManagerExitedLatch.countDown();
-          if (backgroundManagerFuture.isCancelled()) {
-            LOG.info("Background lookup manager exited");
-            LOG.trace(t, "Background lookup manager exited with throwable");
-          } else {
-            LOG.makeAlert(t, "Background lookup manager exited with error!").emit();
-          }
+  private void lookupManagementLoop()
+  {
+    // Sanity check for if we are shutting down
+    if (Thread.currentThread().isInterrupted() || !lifecycleLock.awaitStarted(1, TimeUnit.MILLISECONDS)) {
+      LOG.info("Not updating lookups because process was interrupted or not finished starting yet.");
+      return;
+    }
+
+    final Map<String, Map<String, LookupExtractorFactoryMapContainer>> allLookupTiers = lookupMapConfigRef.get();
+
+    if (allLookupTiers == null) {
+      LOG.info("Not updating lookups because no data exists");
+      return;
+    }
+
+    LOG.debug("Starting lookup sync for on all nodes.");
+
+    final Map<HostAndPort, LookupsStateWithMap> currState = new ConcurrentHashMap<>();
+
+    try {
+      List<ListenableFuture<?>> futures = new ArrayList<>();
```

Comment:
To avoid shared mutable state, suggested:
```java
List<ListenableFuture<Map.Entry<HostAndPort, LookupsStateWithMap>>> futures = new ArrayList<>();
```

```java
          futures.add(
              executorService.submit(() -> {
                try {
                  return new AbstractMap.SimpleImmutableEntry<>(node, doLookupManagementOnNode(node, tierLookups));
                }
                catch (Exception ex) {
                  LOG.makeAlert(
                      ex,
                      "Failed to finish lookup management on node [%s:%s]",
                      node.getHostText(),
                      node.getPort()
                  );
                  return null;
                }
              })
          );
```

```java
        ImmutableMap.Builder<HostAndPort, LookupsStateWithMap> stateBuilder = ImmutableMap.builder();
        allFuture.get(lookupCoordinatorManagerConfig.getAllHostTimeout().getMillis(), TimeUnit.MILLISECONDS)
                 .stream()
                 .filter(java.util.Objects::nonNull)
                 .forEach(stateBuilder::put);
        knownOldState.set(stateBuilder.build());
```